presto-spi/src/main/java/com/facebook/presto/spi/ConnectorSession.java

```diff
@@ -58,4 +58,9 @@ default boolean isReadConstraints()
     {
         return false;
     }
+
+    default WarningCollector getWarningCollector()
+    {
+        return WarningCollector.NOOP;
+    }
```

Comment:
Let's remove default and implement the interface for all its implementations. "Default" is usually discouraged especially for such cases where all implementations have a clear return.